src/grt/src/GlobalRouter.cpp

```diff
@@ -2793,6 +2797,26 @@ void GlobalRouter::makeBtermPins(Net* net,
   }
 }
 
+void GlobalRouter::findClockDrivers()
+{
+  for (Net& net : *_nets) {
+    if (net.getSignalType() == odb::dbSigType::CLOCK) {
+      odb::dbNet* db_net = net.getDbNet();
+      odb::dbITerm* driver = db_net->getFirstOutput();
+      if (driver != nullptr) {
+        std::vector<Pin>& pins = net.getPins();
+        std::vector<Pin>::iterator it = 
```

Comment:
this is really contorted. why are you using find_if instead of find? and the converting the iterator to an index. just use a loop and set the pin while when you find it. 3 lines. 

better yet, set the property when you make the pin instead of rooting around to find it after the fact